packages/core/integration-tests/test/integration/ts-types/exporting/index.ts

```diff
@@ -1,3 +1,8 @@
 export {createMessage} from './message';
 export * from './other';
 export {test as toUpperCase, default} from './test';
+export * as Module1 from './module1';
```

Comment:
Adding `export {hello} from './module1'` here results in the types being duplicated, both outside and inside the `module` block. For strings it's not too bad, but for more complex types might be weird. Maybe we should still hoist the declarations out of the module as before, but create a module like this instead to export them?

```ts
export const hello = "world";
export declare module Module1 {
  export {hello};
}
```

Reply:
That is a good idea. That could probably also include support for the not supported 1) above.
The problem is then that we very likely gets naming conflicts between the hoisted definitions and the exports. Should we generate some random part in the name to avoid these conflicts?

Reply:
I think the hoisting we already do should handle renaming in the top level scope. So you might need something like `export {hello1 as hello}` within the nested module in that case.

Reply:
@devongovett could I ask you to roughly commit to this PR the (failing) tests that describes the intended behavior, then I will try and do the implementation.